controller/binaries.py

```diff
@@ -1,68 +1,57 @@
 import glob
 from pathlib import Path
+from typing import Any, Dict
 
 from termcolor import colored
 
 ROOT_DIR = Path(__file__).parent.parent.resolve()
-FIRMWARES = {
+FIRMWARES: Dict[str, list] = {
     "T1": [],
     "TT": [],
 }
-SUITES = []
 BRIDGES = []
 
 
 def explore(args):
     explore_firmwares(args)
-    explore_suites()
     explore_bridges()
 
 
-def explore_firmwares(args):
-    scan_glob = "firmware/bin/*"
+def explore_firmwares(args: Any) -> None:
+    firmware_binary_files = str(ROOT_DIR / "firmware/bin/*")
     if args.verbosity > 0:
-        print("Scanning {}".format(colored(scan_glob, "yellow")))
-    for fw in glob.glob(scan_glob):
-        if "trezor-emu-core" in fw:
-            version = fw.replace("firmware/bin/trezor-emu-core-v", "")
+        print(f"Scanning {colored(firmware_binary_files, 'yellow')}")
+
+    # Analyzing all potential emulator files and gathering their versions
+    #   (version is located at the end, after emulator identifier and '-v')
+    identifier_TT = "trezor-emu-core"
+    identifier_T1 = "trezor-emu-legacy"
+    for fw in glob.glob(firmware_binary_files):
+        if identifier_TT in fw:
+            version = fw.split(identifier_TT + "-v")[-1]
             FIRMWARES["TT"].append(version)
-            if args.verbosity > 0:
-                print(
-                    "  found {} => {}".format(
-                        colored(fw, "yellow"), colored(version, "magenta")
-                    )
-                )
-        elif "trezor-emu-legacy" in fw:
-            version = fw.replace("firmware/bin/trezor-emu-legacy-v", "")
+        elif identifier_T1 in fw:
+            version = fw.split(identifier_T1 + "-v")[-1]
             FIRMWARES["T1"].append(version)
```

Comment:
What changed is that we are using absolute paths of all binary files (to not being dependent on running this script from the root folder), instead of relative ones from the implicit root folder.

It means that the `fw.replace("firmware/bin/trezor-emu-core-v", "")` would not cut this part from the beginning (as before, when the path was really starting with "firmware/bin..."), but from the middle of the path string, and it would leave the home folder etc., not only the version number